sh.py

```diff
@@ -2491,16 +2491,12 @@ def is_alive(self):
         # so essentially what we're doing is, using this lock, checking if
         # we're calling .wait(), and if we are, let .wait() get the exit code
         # and handle the status, otherwise let us do it.
-        acquired = self._wait_lock.acquire(False)
+        #
+        # Using a small timeout provides backpressure against code that spams
+        # calls to .is_alive() which may block the main thread from acquiring
+        # the lock otherwise.
+        acquired = self._wait_lock.acquire(timeout=0.1)
```

Comment:
I still had to set this back down to:
```suggestion
        acquired = self._wait_lock.acquire(timeout= 0.00001)
```
as it was before https://github.com/amoffat/sh/commit/e9511de5ca9b1d1bf0b42da51fa40073a73fa4d6.

Given we're using the `acquire` provided timeout, is that fine now?